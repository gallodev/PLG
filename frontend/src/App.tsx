import { useEffect, useState, useContext } from 'react';
import { Header } from './components/Header/Header';
import { Accordion } from './components/Accordion/Accordion';
import { CircularProgress } from '@material-ui/core';
import AppContext  from './App.context';
import './App.css';
import { RocketListProps } from './types';
import { IntlProvider } from 'react-intl';

function App() {

  const [rocketData,setRocketData] = useState<any>();
  const [loading,setLoading] = useState<boolean>(true);
  const { api } = useContext(AppContext);

  useEffect(() => {
    const fetchRocketData = async () => {
     const data: RocketListProps = await api.fetchRocketData(); 
     setRocketData(data);
     setLoading(false);
    }

    if(loading) {
      fetchRocketData();  
    }

  },[rocketData, loading]);
  
  return (  
    <IntlProvider locale="pt" defaultLocale="pt">
      <div className="App">    
        <Header />       
        { !loading ?
          <Accordion rocketData={rocketData} />
        : <CircularProgress color="primary" style={{marginTop: 50}}   />}
      </div>
    </IntlProvider>
  );
}

export default App;
