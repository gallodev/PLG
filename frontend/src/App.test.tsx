import { render } from '@testing-library/react';
import App from './App';

describe('[APP] - Should renders app and match snapshot', () => {
  it('[APP] Should render APP and match snapshot', () => {
    const { asFragment } = render(<App />);  
    expect(asFragment()).toMatchSnapshot();
  });
});
