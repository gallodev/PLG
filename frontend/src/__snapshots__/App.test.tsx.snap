// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[APP] - Should renders app and match snapshot [APP] Should render APP and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="sc-gsTEea kjDHvi"
    >
      <div
        class="sc-bdfBQB hhVVxk"
      >
        <ul>
          <li>
            <img
              height="40"
              src="logo-spacex.png"
              width="120"
            />
          </li>
          <li>
            <img
              height="40"
              src="logo-rd-station.png"
              width="120"
            />
          </li>
        </ul>
      </div>
      <h1
        class="sc-dlfnuX hbaWER"
      >
        ðŸš€
      </h1>
    </header>
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px; margin-top: 50px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;
